File: Python3/2274_maximum_consecutive_floors_without_special_floors.py
# https://leetcode.cn/problems/maximum-consecutive-floors-without-special-floors/

class Solution:
    def maxConsecutive(self, bottom, top, special):
        """
        Inputs:
            bottom, top: int
            special: list[int]
        Outputs:
            res: int
        """
        special.sort()
        res = max(special[0] - bottom, top - special[-1])
        for i in range(1, len(special)):
            res = max(res, special[i] - special[i-1] - 1)
        return res

File: Cpp/2274_maximum_consecutive_floors_without_special_floors.cpp
// https://leetcode.cn/problems/minimum-rounds-to-complete-all-tasks/

class Solution {
public:
    int maxConsecutive(int bottom, int top, vector<int>& special) {
        sort(special.begin(), special.end());
        int res = max(special[0] - bottom, top - special.back());
        for (int i = 1; i < special.size(); ++i) {
            res = max(res, special[i] - special[i-1] - 1);
        }
        return res;
    }
};
